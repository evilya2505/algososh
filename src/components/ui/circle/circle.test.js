import React from "react";
import renderer from "react-test-renderer";
import { Circle } from "./circle";

describe("Компонент Circle", () => {
  it("рендер элемента без буквы", () => {
    const tree = renderer.create(<Circle />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("рендер элемента с буквами", () => {
    const tree = renderer.create(<Circle letter="A" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("рендер элемента с head", () => {
    const tree = renderer.create(<Circle head="H" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("рендер элемента с react-элементом в head", () => {
    const tree = renderer
      .create(<Circle head={<div>React Element</div>} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("рендер элемента с tail", () => {
    const tree = renderer.create(<Circle tail="T" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("рендер элемента с react-элементом в tail", () => {
    const tree = renderer
      .create(<Circle tail={<div>React Element</div>} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("рендер элемента с index", () => {
    const tree = renderer.create(<Circle index={5} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("рендер элемента с пропом isSmall === true", () => {
    const tree = renderer.create(<Circle isSmall={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("рендер элемента в состоянии default", () => {
    const tree = renderer.create(<Circle state="default" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("рендер элемента в состоянии changing", () => {
    const tree = renderer.create(<Circle state="changing" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("рендер элемента в состоянии modified", () => {
    const tree = renderer.create(<Circle state="modified" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
