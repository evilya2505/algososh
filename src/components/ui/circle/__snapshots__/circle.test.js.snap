// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент Circle рендер элемента без буквы 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Компонент Circle рендер элемента в состоянии changing 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Компонент Circle рендер элемента в состоянии default 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Компонент Circle рендер элемента в состоянии modified 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Компонент Circle рендер элемента с head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    H
  </div>
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Компонент Circle рендер элемента с index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    5
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Компонент Circle рендер элемента с react-элементом в head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div>
      React Element
    </div>
  </div>
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Компонент Circle рендер элемента с react-элементом в tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div>
      React Element
    </div>
  </div>
</div>
`;

exports[`Компонент Circle рендер элемента с tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    T
  </div>
</div>
`;

exports[`Компонент Circle рендер элемента с буквами 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      A
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Компонент Circle рендер элемента с пропом isSmall === true 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
